frontend/src/components/teacher/session-creation/steps/ChooseClass.tsx

```diff
@@ -1,16 +1,146 @@
+            "active"
+          ) {
+            activeAssessments += 1;
+          }
+        });
+
+        return {
+          ...classCard,
+          activeAssessments,
+          assessmentCount: testSessions.length,
+          studentCount: students.length,
+        };
+      },
+    );
+  }, [data]);
+
+  const { paginatedData, totalPages, currentPage, setCurrentPage } =
+    usePaginatedData(classCards);
+
   return (
-    <Text color="blue.300" textAlign="left" textStyle="header4">
-      Choose a Classroom
-    </Text>
+    <VStack align="left" spacing="2">
+      <Text color="blue.300" textAlign="left" textStyle="header4">
+        Choose a Classroom
+      </Text>
+      <Text color="grey.300" textStyle="paragraph">
+        Please choose classrooms you want this assessment to be distributed to.
+      </Text>
+      <Box pt="6">
+        {loading && (
+          <Center flex="1" margin="15%">
+            <LoadingState />
+          </Center>
+        )}
+        {error && (
+          <Box height="100%" mt={10}>
+            <ErrorState />
+          </Box>
+        )}
+        {classCards?.length !== 0 ? (
+          <Box alignItems="left">
+            <Flex flexWrap="wrap">
+              {paginatedData?.map(
+                ({
+                  id,
+                  activeAssessments,
+                  assessmentCount,
+                  gradeLevel,
+                  className,
+                  startDate,
+                  studentCount,
+                }) => (
+                  <Flex
+                    key={id}
+                    cursor="pointer"
+                    onClick={() => setClassId(id)}
+                    paddingRight="4"
+                    paddingTop="4"
+                  >
+                    <ClassroomCard
+                      key={id}
+                      activeAssessments={activeAssessments}
+                      assessmentCount={assessmentCount}
+                      disabledOptions={true}
+                      grade={gradeLevel}
+                      id={id}
+                      name={className}
+                      selected={selectedClassId === id}
+                      startDate={startDate}
+                      studentCount={studentCount}
+                    />
+                  </Flex>
+                ),
+              )}
+            </Flex>
+            <VStack
+              alignItems="center"
+              paddingBottom="6"
+              paddingTop="6"
+              spacing="6"
+              width="100%"
+            >
+              {totalPages > 1 && (
+                <Pagination
+                  currentPage={currentPage}
+                  onPageChange={setCurrentPage}
+                  pagesCount={totalPages}
+                />
+              )}
+            </VStack>
+          </Box>
+        ) : (
+          <EmptyDistributeClassroomsMessage onClick={navigateToClassrooms} />
+        )}
+      </Box>
+    </VStack>
```

Comment:
some of this is duplicate from `ClassroomPages`, but I didn't think it was worth separating into a parent component (since we only use this pattern twice for now), but let me know if you think we should. Or, if there are some ways we can separate things into common components.

Reply:
hmm yea there's a lot of minor differences that'll make the code ugly so i think fine as is